habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDependenciesMojo.java

```diff
@@ -0,0 +1,156 @@
+                + "/venv-support/" + this.anchorSourceDirectory.getName());
+        Path srcRoot = this.anchorSourceDirectory.toPath();
+        Path destRoot = this.anchorOutputDirectory.toPath();
+
+        Map<Path, FileSet> dependencyFileSets = new HashMap<>();
+        for (MavenProject project : habushuProjects) {
+            Path projectPath = project.getFile().getParentFile().toPath();
+            Path relativeProjectPath = srcRoot.relativize(projectPath);
+            FileSet fileSet = getDefaultFileSet();
+            fileSet.setDirectory(projectPath.toString());
+            dependencyFileSets.put(relativeProjectPath, fileSet);
+        }
+
+        FileSetManager fileSetManager = new FileSetManager();
+        for (Path project : dependencyFileSets.keySet()) {
+            FileSet fileSet = dependencyFileSets.get(project);
+            logger.info("Copying {} files from {}",
+                    fileSetManager.getIncludedFiles(fileSet).length,
+                    project.getFileName()
+            );
+            for (String includedFile : fileSetManager.getIncludedFiles(fileSet)) {
+                Path relativePath = project.resolve(includedFile);
+                Files.createDirectories(destRoot.resolve(relativePath).getParent());
+                Files.copy(srcRoot.resolve(relativePath), destRoot.resolve(relativePath));
+            }
+        }
+    }
+
+    private FileSet getDefaultFileSet() {
+        FileSet fileSet = new FileSet();
+        fileSet.addExclude("target/**");
+        fileSet.addExclude(".venv/**");
+        fileSet.addExclude("dist/**");
+        return fileSet;
+    }
+
+    /**
+     * Checks listed habushu-type dependencies against the set of projects included in the Maven build's session
+     * @return the corresponding Maven projects that match the habushu-type dependencies
+     */
+    protected Set<MavenProject> getHabushuProjects() {
+        HashSet<MavenProject> habushuProjects = new HashSet<>();
+        collectHabushuDependenciesAsProjects(project, habushuProjects);
+        return habushuProjects;
+    }
+
+    /**
+     * Collects the projects of the habushu-type dependencies and adds them to the given project set
+     * @param currentProject the project to interrogate the habushu-type dependencies against
+     * @param habushuProjects the set to append to with matching projects
+     */
+    private void collectHabushuDependenciesAsProjects(MavenProject currentProject, Set<MavenProject> habushuProjects) {
+        Set<String> habushuDeps = currentProject.getDependencies().stream()
+                .filter(d -> HABUSHU.equals(d.getType()))
+                .map(ContainerizeDependenciesMojo::toGav)
+                .collect(Collectors.toSet());
+        for (MavenProject project : getSession().getProjects()) {
+            if (habushuDeps.contains(toGav(project))) {
+                logger.info("Found habushu-type dependency's project {}.", project);
```

Comment:
`Found habushu-type dependency's project {}.` -> `Found habushu-type dependency's within project {}.`

Reply:
Or should it be `Found project for habushu-type dependency`?

Reply:
second one is better to me, updated